lib/player.js

```diff
@@ -3854,12 +3853,15 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           const stringData = await this.getTextData_(uri,
               this.networkingEngine_,
               this.config_.streaming.retryParameters);
-          const SrtTextParser = shaka.text.SrtTextParser;
-          const vvtText = SrtTextParser.srt2webvtt(stringData);
+          const vvtText = this.convertToWebVTT_(stringData, mimeType);
           const blob = new Blob([vvtText], {type: 'text/vtt'});
           uri = URL.createObjectURL(blob);
           mimeType = 'text/vtt';
-        } catch (error) {}
+        } catch (error) {
+          if (error instanceof shaka.util.Error) {
+            throw error;
+          }
+        }
```

Comment:
I didn't notice this in my last review, but is it intentional that this should only rethrow shaka.util.Error types, but ignore other errors and fall through to the mimeType check and TEXT_ONLY_WEBVTT_SRC_EQUALS error type?

Reply:
Yes, it’s intentionally. If it is not a Shaka error throw a Shaka error